### PR TITLE
Closes #48 file review and catch up to current admiral

### LINKED ISSUE #48: General Issue: Review content and update as needed
### Background Information

We developed this template repo a year and some of the content is outdated.  

A full review is needed and updates/discussion can either be made with the issue tagged or additional issues can be created.

### Definition of Done

- [ ] Thorough review of Files
- [ ] Update relevant files

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
Here is a couple of other points we may need to address:

- [x] renv package versions not perfectly aligned with development version, e.g. profile for R v4.2.0 devtools & rlang are not harmonzied
- [x] does _pkgdown.yml need a more generic look?
- [x] should be documented somewhere the _pkgdown.yml file is closely tied to ./man/roxygen/meta.R
- [x] CODEOWNERS file currently exists, considering we don't have a defined implementation is this a good idea?
- [x] Need extra eyes on .gitignore, .Rbuildignore, .Rprofile
- [x] WORDLIST file probably varies across packages, do we need to come up with any standardization plan besides append as necessary?
- [x] dyn_links for README don't all work

**Comment by bms63** [comment]
@zdz2101 I add you to the repo

> Here is a couple of other points we may need to address:
> 
> * renv package versions not perfectly aligned with development version, e.g. profile for R v4.2.0 devtools & rlang are not harmonzied

Do we just need the latest propagation from admiralci?

> * does _pkgdown.yml need a more generic look?

I don't think so - looks pretty generic t o me.

> * should be documented somewhere the _pkgdown.yml file is closely tied to ./man/roxygen/meta.R

Sounds good!  Maybe the reference file for hello_admiral could talk about that in the Details section and link out to the admiraldev pages around keywords/family

> * CODEOWNERS file currently exists, considering we don't have a defined implementation is this a good idea?

I think we could add a comment in the github section saying what this is doing and put a link in there for admiralci?

image

> * Need extra eyes on .gitignore, .Rbuildignore, .Rprofile

Could you list out what are in these files and we can review what can be thrown out/kept?

> * WORDLIST file probably varies across packages, do we need to come up with any standardization plan besides append as necessary?

I think this is being propagated now from admiralci?  @galachad ?

> * dyn_links for README don't all work

Can you fix please? We will need a note saying this needs to change to their respective pkacage/repo name?

**Comment by zdz2101** [comment]
`.gitignore` here:
image

`.RBuildignore` here: 
image

`.Rprofile` here: 
image

**Comment by galachad** [comment]
WORDLIST is not propagated across repositories. The WORDLIST contain exclude words from current package and can be easily updated with `spelling::update_wordlist()`. If there any new mistake shows up, then the spell check job will failed.

File: NAMESPACE
Comment (bms63) [inline NAMESPACE:6]: ohh we got rid of this package - lets remove it here as well

Comment (bms63) [inline NAMESPACE]: ohh we got rid of this package - lets remove it here as well

**Review by bms63 — COMMENTED** [review]
I think we should remove the Onbaording Template from the `.github` folder as well

**Comment by bms63** [comment]
Lets get rid of all these

image

**Comment by bms63** [comment]
I think stuff in the `.gitignore` is fine.

**Comment by bms63** [comment]
This reference to the `pre-release` can get removed 
image

Do we have anything on setting up the code coverage branch?

**Comment by zdz2101** [comment]
> This reference to the `pre-release` can get removed image
> 
> Do we have anything on setting up the code coverage branch?

We have a branch specifically for code coverage? I thought this was handled by this chunk in the `common.yml`, speaking of I think I need to update this as well, seems to be using older versions of R:
image

**Comment by bms63** [comment]
check this out - actually be cool to set this up for `admiraltemplate` 
image

**Comment by zdz2101** [comment]
> check this out - actually be cool to set this up for `admiraltemplate` image

https://github.com/pharmaverse/admiraltemplate/tree/badges Is it already set up?

**Comment by bms63** [comment]
Well...I stand corrected.  Can we get the test coverage to be 100%.  Also this table shoudln't be there


image

File: README.md
Comment (zdz2101) [inline README.md:1]: @bms63 that table should go away with this deletion

File: README.md
Comment (zdz2101) [inline README.md:5]: @bms63 that table should go away with this deletion

**Comment by bms63** [comment]
what about just releasing the package to CRAN as well - want to be a package maintainer @zdz2101 ?  that way we get the proper badges.  

Also - maybe a proper Changelog is in order for each release.  probably just one a year or whenever.

**Comment by zdz2101** [comment]
> what about just releasing the package to CRAN as well - want to be a package maintainer @zdz2101 ? that way we get the proper badges.
> 
> Also - maybe a proper Changelog is in order for each release. probably just one a year or whenever.

Sure, I can do that

**Comment by bms63** [comment]
Can you fix the failing check.  Also - maybe we should not load the package to CRAN as it clutters up the admiral landscape.  I would like to see a changelog/news updated with a 0.2 notes section and an updated Description File

**Comment by zdz2101** [comment]
> Can you fix the failing check. Also - maybe we should not load the package to CRAN as it clutters up the admiral landscape. I would like to see a changelog/news updated with a 0.2 notes section and an updated Description File

For the news, do you have a full detailed list of all the modifications made aka file by file?

**Comment by bms63** [comment]
> > Can you fix the failing check. Also - maybe we should not load the package to CRAN as it clutters up the admiral landscape. I would like to see a changelog/news updated with a 0.2 notes section and an updated Description File
> 
> For the news, do you have a full detailed list of all the modifications made aka file by file?

I'm just thinking of a general feel of what updates were made.  Lock Files updated,  Links fixed, Templates brought up to current admiral standards.  Perhaps rather than linking the Issues in the NEws, we can just link the PR since it was a bit of grab bag

File: .github/workflows/common.yml
Comment (bms63) [inline .github/workflows/common.yml]: can we remove references to `pre-release`

File: DESCRIPTION
Comment (bms63) [inline DESCRIPTION]: So we have some compatibility issues - we should have admiral (>= 0.10.0) and admiraldev (>=0.3).

File: NAMESPACE
Comment (bms63) [inline NAMESPACE]: ummm...do we want to have the whole package or just bits and bobs??

File: DESCRIPTION
Comment (bms63) [inline DESCRIPTION]: i think we just call it `admiraltemplate` here and in other parts of the documenation

File: NEWS.md
Comment (bms63) [inline NEWS.md:8]: oohhh could you put in the stuff about the details you updated in the `my_first_fcn()`.  I think being able to see this is nice.

Comment (bms63) [reply NEWS.md:8]: ah...i see it in next section and this isn't an update to the function...scratch it

File: R/my_first_fcn.R
Comment (bms63) [inline R/my_first_fcn.R]: We should remove authors as we don't use that anymore

**Review by bms63 — CHANGES_REQUESTED** [review]
@rossfarrugia we are working on aligning the template.  If you have any input feel free to drop in anything, but don't feel like you need to review (but we always welcome your insights)

File: .github/workflows/common.yml
Comment (zdz2101) [reply .github/workflows/common.yml]: Should I actually just copy over the entire [admiralci/.github/workflows](https://github.com/pharmaverse/admiralci/tree/main/.github/workflows) directory? It looks like using the `common.yml` is out of date?

File: NAMESPACE
Comment (zdz2101) [reply NAMESPACE]: I think it could go either way, in `admiral` we import the whole thing, in something like `admiralvaccine` they use bits and blobs, went ahead and copied the bits and blobs over to demonstrate?

File: DESCRIPTION
Comment (zdz2101) [reply DESCRIPTION]: Think I covered this with the most recent commit

File: .github/workflows/common.yml
Comment (bms63) [reply .github/workflows/common.yml]: Yeah that might be best

**Review by bms63 — APPROVED** [review]
I guess the only thing is the common.yml file.  After this do you mind making an announcement in slack that we have updated our template and that people can check it out for possible package extensions??

File: .github/workflows/common.yml
Comment (rossfarrugia) [reply .github/workflows/common.yml]: @zdz2101 @bms63 i might be missing the point here and sorry if so, but wasn't the whole point of using `common.yml` that we don't ever need to copy the individual workflows over from admiralci? Tagging @bundfussr @thomas-neitmann here to comment just in case, as not sure i understand this suggested update

**Comment by rossfarrugia** [comment]
> I guess the only thing is the common.yml file. After this do you mind making an announcement in slack that we have updated our template and that people can check it out for possible package extensions??

@zdz2101 @bms63 can we be careful on the messaging around package extensions on public slack? its not that this is open game for anyone to go create these. (of course they can do - as this is OS and all permissively licensed so anyone can do what they like) but we want this to be structured and go through us, so if anyone has an idea for a new extension they should contact us first. we then do things like ensure someone from our core dev team can be on hand to help ensure they follow our conventions. please make that clear in any comms. i think i said it once at a call before, but imagine if the 4th and 5th MacDonalds stores had served terrible food then they could have brought down the whole franchise before they really ever got off the ground.

**Comment by rossfarrugia** [comment]
> @rossfarrugia we are working on aligning the template. If you have any input feel free to drop in anything, but don't feel like you need to review (but we always welcome your insights)

I did a quick sanity check and left just one comment reply above

**Comment by rossfarrugia** [comment]
> @rossfarrugia we are working on aligning the template. If you have any input feel free to drop in anything, but don't feel like you need to review (but we always welcome your insights)

I did a quick sanity check and left just one comment reply above

File: .github/workflows/common.yml
Comment (bms63) [reply .github/workflows/common.yml]: I think he is just talking about what is in admiral workflow folder - which is common, cran status (leave that one alone) and template check 
image

Comment (rossfarrugia) [reply .github/workflows/common.yml]: i'm not sure what's the difference between `templates.yml` in admiral and `check-templates.yml` in admiralci? my point was more not to copy over all the admiralci workflows here, as they should live only there and then we reference them in each package ext through the `common.yml` - as per https://pharmaverse.github.io/admiralci/main/index.html#reuse-recommended

**Review by bms63 — COMMENTED** [review]
Can we look into adopting testthat 3rd edition as well @zdz2101

File: .github/workflows/common.yml
Comment (zdz2101) [reply .github/workflows/common.yml]: Yes I definitely was a bit confusing there, I did indeed mean `admiral` not `admiralci`. I went ahead and added the `templates.yml` which is a `.yml` that triggers the  Check-Templates action after a reviewer approves a PR. As for the `common.yml` I updated to remove the mention of `pre-release` , speaking of which, we need to update for `admiral` and `admiraldev`. I can go ahead and make issues for that too (good first PR)

Comment (rossfarrugia) [reply .github/workflows/common.yml]: great, thanks Zelos!

**Review by bms63 — APPROVED** [review]
I'm excited!

**Comment by bms63** [comment]
Please merge @zdz2101 if you think this is completed!!  Thank you for bringing this up to speed.  

Wondering how we can keep this current - maybe at each release we should have a checkin of the template???

**Pull request merged**
